tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/ExportedRuleTests.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Rules
+{
+    public sealed class RuleExporterTests : XamlRuleTestBase
+    {
+        [Theory]
+        [MemberData(nameof(GetAllExportedMembers))]
+        public void ExportedRulesMustBeStaticFields(string name, MemberInfo member)
+        {
+            Assert.True(member is FieldInfo field && field.IsStatic, $"'{name}' must be a static field.");
+        }
+
+        [Theory]
+        [MemberData(nameof(GetAllExportedMembers))]
+        public void ExportedRulesMustHaveAppliesTo(string name, MemberInfo member)
+        {
+            var attribute = member.GetCustomAttribute<AppliesToAttribute>();
+
+            Assert.True(attribute != null, $"'{name}' must be marked with AppliesTo");
+        }
+
+        [Theory]
+        [MemberData(nameof(GetAllExportedMembers))]
+        public void ExportedRulesMustExist(string name, MemberInfo member)
+        {
+            var attribute = member.GetCustomAttribute<ExportPropertyXamlRuleDefinitionAttribute>();
+
+            Assert.NotNull(attribute);
+
+            var assembly = Assembly.Load(attribute.XamlResourceAssemblyName);
+
+            using Stream stream = assembly.GetManifestResourceStream(attribute.XamlResourceStreamName);
+
+            Assert.True(stream != null, $"The rule '{attribute.XamlResourceStreamName}' indicated by '{name}' does not exist in assembly '{attribute.XamlResourceAssemblyName}'.");
+        }
+
+        public static IEnumerable<object[]> GetAllExportedMembers()
+        {
+            foreach (var member in typeof(RuleExporter).GetMembers())
+            {
+                if (member.DeclaringType == typeof(RuleExporter))
```

Comment:
Could pass `BindingFlags.DeclaredOnly` to `GetMembers` instead of this.

Reply:
And `BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public`, this ended up being much simpiler.